plugins/modules/api.py

```diff
@@ -358,7 +360,7 @@ def list_to_dic(self, ldict):
         for p in ldict:
             if '=' not in p:
                 self.errors("missing '=' after '%s'" % p)
-            p = p.split('=')
+            p = p.split('=', 1)
             if p[1]:
```

Comment:
example code:
```
ldict = "comment='test ansible rolemodule' disabled=no netmask=255.255.255.0 network=192.168.192.0 address=192.168.192.20 interface=ether2"


def list_to_dic(ldict):
    dict = {}
    print(ldict)
    for p in ldict:
        if '=' in p:
            p = p.split('=', 1)
            dict[p[0]] = p[1]
    return dict

print(list_to_dic(ldict.split(' ')))
```

test:
```

["comment='test", 'ansible', "rolemodule'", 'disabled=no', 'netmask=255.255.255.0', 'network=192.168.192.0', 'address=192.168.192.20', 'interface=ether2']

{'comment': "'test", 'disabled': 'no', 'netmask': '255.255.255.0', 'network': '192.168.192.0', 'address': '192.168.192.20', 'interface': 'ether2'}
```

let me explain 

The idea behind list_to_dic() is to translate ansible "single string" parameter to proper key=value pair for librouteros https://librouteros.readthedocs.io/en/latest/path.html#add

As general we cannot make a list/dict of all router os parameters but always key=value is required 
Since we split ansible parameter in earyly stage, for example 
```
    def api_add(self):
        param = self.list_to_dic(self.add.split(' '))
```

adding p = p.split('=', 1) maxsplit limit is not really usefull (check test: results) 

So additional work is needed in order to fix that, I like the approuch with shlex in #45 but this code must be tested cearfully for example arbitary cmd..!

`if p[1]:`

Prevent empty key's .. however routeros will return msg for "invalid value for argument .." so I guess will be better without it !

For WHERE change .. to be tested with `def api_query(self):` 

Note 'api_query(self):'  with this function we should translate ansible parameter "single string" to python code .. what I try to say .. at the moment this function introduce a limitation to what we can do with librouteros: https://librouteros.readthedocs.io/en/latest/query.html and need improvments

Reply:
example code:
```
ldict = "comment='test ansible rolemodule' disabled=no netmask=255.255.255.0 network=192.168.192.0 address=192.168.192.20 interface=ether2"


def list_to_dic(ldict):
    dict = {}
    print(ldict)
    for p in ldict:
        if '=' in p:
            p = p.split('=', 1)
            dict[p[0]] = p[1]
    return dict

print(list_to_dic(ldict.split(' ')))
```

test:
```

["comment='test", 'ansible', "rolemodule'", 'disabled=no', 'netmask=255.255.255.0', 'network=192.168.192.0', 'address=192.168.192.20', 'interface=ether2']

{'comment': "'test", 'disabled': 'no', 'netmask': '255.255.255.0', 'network': '192.168.192.0', 'address': '192.168.192.20', 'interface': 'ether2'}
```

let me explain 

The idea behind list_to_dic() is to translate ansible "single string" parameter to proper key=value pair for librouteros https://librouteros.readthedocs.io/en/latest/path.html#add

As general we cannot make a list/dict of all router os parameters but always key=value is required 
Since we split ansible parameter in earyly stage, for example 
```
    def api_add(self):
        param = self.list_to_dic(self.add.split(' '))
```

adding p = p.split('=', 1) maxsplit limit is not really usefull (check test: results) 

So additional work is needed in order to fix that, I like the approuch with shlex in #45 but this code must be tested cearfully for example arbitary cmd..!

`if p[1]:`

Prevent empty key's .. however routeros will return msg for "invalid value for argument .." so I guess will be better without it !

For WHERE change .. to be tested with `def api_query(self):` 

Note 'api_query(self):'  with this function we should translate ansible parameter "single string" to python code .. what I try to say .. at the moment this function introduce a limitation to what we can do with librouteros: https://librouteros.readthedocs.io/en/latest/query.html and need improvments

Reply:
Ok, so basically the `if` is needed because command splitting is currently broken.

But what about commands which accept something that's not in the `a=b` form? Like `sign` in `/certificate` (https://help.mikrotik.com/docs/display/ROS/Certificates#Certificates-SignCertificate)?

> adding p = p.split('=', 1) maxsplit limit is not really usefull (check test: results)

It is! Have you tried something like `comment='a=b'`? Without the maxsplit, it will put `"comment": "'a"` into the dictionary, instead of `"comment": "'a=b'"`.

Reply:
arbitrary cmd do the job, example: https://github.com/NikolayDachev/ansible_collections/blob/master/nikolaydachev/routeros_api/roles/ros_certificates/tasks/sign.yml

where we don't have **key=value** we can use _**number**_=**value**

note: **arbitrary** and **add** are different for librouteros 

https://github.com/luqasz/librouteros/issues/117
The hint is: **ip service print detail**  or check any other router os cmd